Assets/Scripts/Model/Types/Coordinate.cs

```diff
@@ -27,14 +27,30 @@ public static Coordinate Create(int[] coordinates)
 
         }
 
-        public static Coordinate Create(int x, int y)
+        public override bool Equals(Object obj)
         {
-            return new Coordinate(x, y);
+           if (! (obj is Coordinate)) return false;
+
+           Coordinate p = (Coordinate)obj;
+           return X == p.X & Y == p.Y;
         }
 
-        public bool Equals(Coordinate other)
+        public override int GetHashCode()
         {
-            return other != null && other.X == this.X && other.Y == this.Y;
+            return ShiftAndWrap(X.GetHashCode(), 2) ^ Y.GetHashCode();
```

Comment:
Make this a more easily understandable hashcode. No point having it use a generic complex method when something like y*1000+x will suffice for our purposes.

Reply:
Done. Opted for 10000.